js/ui/lookup.js

```diff
@@ -489,47 +489,99 @@ const Lookup = DropDownList.inherit({
         }
     },
 
-    _setPopupPosition: function() {
-        if(!this.option('itemCenteringEnabled')) return;
+    _getDifferenceOffsets: function(selectedListItem) {
+        return selectedListItem.offset().top - $(this.element()).offset().top;
+    },
 
+    _isCenteringEnabled(index, count) {
+        return 1 < index && index < (count - 2);
+    },
 
-        const selectedIndex = this._list.option('selectedIndex');
-        const flipped = this._popup._$wrapper.hasClass(LOOKUP_POPOVER_FLIP_VERTICAL_CLASS);
-        if(selectedIndex === -1 || flipped) return;
+    _getPopupOffset: function(selectedIndex) {
+        const listItemsCount = this._list.option('items').length;
+
+        if(listItemsCount === 0) return;
 
         const selectedListItem = $(this._list.element()).find('.' + LIST_ITEM_SELECTED_CLASS);
+        const differenceOfHeights = (selectedListItem.height() - $(this.element()).height()) / 2;
+        const lookupOffset = $(this._list.element()).offset().top;
+        const dropDownHeightOption = this.option('dropDownOptions.height');
+        const popupHeight = (typeof dropDownHeightOption === 'function') ? dropDownHeightOption() : dropDownHeightOption;
+        const windowHeight = $(window).height();
+
+        let offsetTop = 0;
+
+        if(selectedIndex !== -1) {
+            if(this._isCenteringEnabled(selectedIndex, listItemsCount)) {
+                this._scrollToSelectedItem();
+
+                const scrollOffsetTop = (popupHeight - selectedListItem.height()) / 2 - this._getDifferenceOffsets(selectedListItem);
+
+                this._list.scrollTo(this._list.scrollTop() + MATERIAL_LOOKUP_LIST_PADDING / 2 - scrollOffsetTop);
+
+                offsetTop = differenceOfHeights + this._getDifferenceOffsets(selectedListItem);
```

Comment:
It seems that this line is need in all condition branches

Reply:
Yes, but  this._getDifferenceOffsets(selectedListItem) calculate after scroll each time

Reply:
And? Can we calculate it once after all scrollings?

Reply:
As you can see in the first case this._getDifferenceOffsets(selectedListItem) calls twice. The second time is based on the previous value of this._getDifferenceOffsets(selectedListItem)

Reply:
I can extract this line from the other two cases